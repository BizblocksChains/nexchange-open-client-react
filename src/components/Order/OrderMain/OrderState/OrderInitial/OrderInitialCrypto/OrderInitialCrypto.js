import React, { Component } from 'react';
import CopyToClipboard from 'react-copy-to-clipboard';
import i18n from '../../../../../../i18n';
import { I18n } from 'react-i18next';
import styles from '../OrderInitial.scss';

class OrderInitial extends Component {
  triggerCopyTooltip() {
    $('#copy-to-clipboard').tooltip({
      trigger: 'click',
      placement: 'top',
    });

    $('#copy-to-clipboard')
      .tooltip('hide')
      .attr('data-original-title', i18n.t('order.copy'))
      .tooltip('show');

    setTimeout(() => {
      $('#copy-to-clipboard').tooltip('destroy');
    }, 1000);
  }

  getDepositAddressQr() {
    return `https://chart.googleapis.com/chart?chs=250x250&chld=L|2&cht=qr&chl=
			${this.props.order.deposit_address.address}`;
  }

  render() {
    return (
      <I18n ns="translations">
        {t => (
          <div className={styles.container}>
            <div className={styles['qr-container']}>
              <img className={styles.qr} src={this.getDepositAddressQr()} alt={t('order.qr')} />
            </div>

            <div className={styles.details}>
              <h3>
                {t('order.initial1')}:{' '}
                <span className={styles.time}>
                  <b>{this.props.time}</b>
                </span>
              </h3>

              <h4>
                {t('order.initial2')}{' '}
                <b>
                  {this.props.order.amount_quote} {this.props.order.pair.quote.code}
                </b>{' '}
                {t('order.initial3')}
                <br />
                <b className={styles.address} style={{ wordWrap: 'break-word' }}>
                  {this.props.order.deposit_address.address}
                </b>
              </h4>

              <CopyToClipboard text={this.props.order.deposit_address.address} onCopy={() => this.triggerCopyTooltip()}>
                <button id="copy-to-clipboard" type="button" className="btn btn-default" data-test="copy-address">
                  {t('order.initial4')}
                </button>
              </CopyToClipboard>
            </div>
          </div>
        )}
      </I18n>
    );
  }
}

export default OrderInitial;
