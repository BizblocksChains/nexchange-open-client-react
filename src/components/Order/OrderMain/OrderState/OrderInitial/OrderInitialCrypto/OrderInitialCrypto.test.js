import React from 'react';
import { shallow } from 'enzyme';
import OrderInitialCrypto from './OrderInitialCrypto';
import order from 'Mocks/order';

describe('OrderInitialCrypto', () => {
  it('renders correctly', () => {
    const wrapShallow = shallow(<OrderInitialCrypto order={order} />).dive();
    expect(wrapShallow).toMatchSnapshot();
  });
});
