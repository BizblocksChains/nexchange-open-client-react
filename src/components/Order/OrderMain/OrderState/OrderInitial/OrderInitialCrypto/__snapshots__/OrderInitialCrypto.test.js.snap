// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderInitialCrypto renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="qr-container"
>
          <img
                    alt="order.qr"
                    className="qr"
                    src="https://chart.googleapis.com/chart?chs=250x250&chld=L|2&cht=qr&chl=
          			0x6ce146d0431079fe70dffee5fb15ba687d7f9ae7"
          />
</div>,
        <div
          className="details"
>
          <h3>
                    order.initial1
                    :
                     
                    <span
                              className="time"
                    >
                              <b />
                    </span>
          </h3>
          <h4>
                    order.initial2
                     
                    <b>
                              0.10000000
                               
                              ETH
                    </b>
                     
                    order.initial3
                    <br />
                    <b
                              className="address"
                              style={
                                        Object {
                                                  "wordWrap": "break-word",
                                                }
                              }
                    >
                              0x6ce146d0431079fe70dffee5fb15ba687d7f9ae7
                    </b>
          </h4>
          <CopyToClipboard
                    onCopy={[Function]}
                    options={undefined}
                    text="0x6ce146d0431079fe70dffee5fb15ba687d7f9ae7"
          >
                    <button
                              className="btn btn-default"
                              data-test="copy-address"
                              id="copy-to-clipboard"
                              type="button"
                    >
                              order.initial4
                    </button>
          </CopyToClipboard>
</div>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="order.qr"
            className="qr"
            src="https://chart.googleapis.com/chart?chs=250x250&chld=L|2&cht=qr&chl=
			0x6ce146d0431079fe70dffee5fb15ba687d7f9ae7"
/>,
          "className": "qr-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "order.qr",
            "className": "qr",
            "src": "https://chart.googleapis.com/chart?chs=250x250&chld=L|2&cht=qr&chl=
			0x6ce146d0431079fe70dffee5fb15ba687d7f9ae7",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              order.initial1
              :
               
              <span
                            className="time"
              >
                            <b />
              </span>
</h3>,
            <h4>
              order.initial2
               
              <b>
                            0.10000000
                             
                            ETH
              </b>
               
              order.initial3
              <br />
              <b
                            className="address"
                            style={
                                          Object {
                                                        "wordWrap": "break-word",
                                                      }
                            }
              >
                            0x6ce146d0431079fe70dffee5fb15ba687d7f9ae7
              </b>
</h4>,
            <CopyToClipboard
              onCopy={[Function]}
              options={undefined}
              text="0x6ce146d0431079fe70dffee5fb15ba687d7f9ae7"
>
              <button
                            className="btn btn-default"
                            data-test="copy-address"
                            id="copy-to-clipboard"
                            type="button"
              >
                            order.initial4
              </button>
</CopyToClipboard>,
          ],
          "className": "details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "order.initial1",
                ":",
                " ",
                <span
                  className="time"
>
                  <b />
</span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "order.initial1",
              ":",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <b />,
                  "className": "time",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "b",
                },
                "type": "span",
              },
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "order.initial2",
                " ",
                <b>
                  0.10000000
                   
                  ETH
</b>,
                " ",
                "order.initial3",
                <br />,
                <b
                  className="address"
                  style={
                                    Object {
                                                      "wordWrap": "break-word",
                                                    }
                  }
>
                  0x6ce146d0431079fe70dffee5fb15ba687d7f9ae7
</b>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "order.initial2",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "0.10000000",
                    " ",
                    "ETH",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "0.10000000",
                  " ",
                  "ETH",
                ],
                "type": "b",
              },
              " ",
              "order.initial3",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "0x6ce146d0431079fe70dffee5fb15ba687d7f9ae7",
                  "className": "address",
                  "style": Object {
                    "wordWrap": "break-word",
                  },
                },
                "ref": null,
                "rendered": "0x6ce146d0431079fe70dffee5fb15ba687d7f9ae7",
                "type": "b",
              },
            ],
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <button
                className="btn btn-default"
                data-test="copy-address"
                id="copy-to-clipboard"
                type="button"
>
                order.initial4
</button>,
              "onCopy": [Function],
              "options": undefined,
              "text": "0x6ce146d0431079fe70dffee5fb15ba687d7f9ae7",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "order.initial4",
                "className": "btn btn-default",
                "data-test": "copy-address",
                "id": "copy-to-clipboard",
                "type": "button",
              },
              "ref": null,
              "rendered": "order.initial4",
              "type": "button",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="qr-container"
>
            <img
                        alt="order.qr"
                        className="qr"
                        src="https://chart.googleapis.com/chart?chs=250x250&chld=L|2&cht=qr&chl=
            			0x6ce146d0431079fe70dffee5fb15ba687d7f9ae7"
            />
</div>,
          <div
            className="details"
>
            <h3>
                        order.initial1
                        :
                         
                        <span
                                    className="time"
                        >
                                    <b />
                        </span>
            </h3>
            <h4>
                        order.initial2
                         
                        <b>
                                    0.10000000
                                     
                                    ETH
                        </b>
                         
                        order.initial3
                        <br />
                        <b
                                    className="address"
                                    style={
                                                Object {
                                                            "wordWrap": "break-word",
                                                          }
                                    }
                        >
                                    0x6ce146d0431079fe70dffee5fb15ba687d7f9ae7
                        </b>
            </h4>
            <CopyToClipboard
                        onCopy={[Function]}
                        options={undefined}
                        text="0x6ce146d0431079fe70dffee5fb15ba687d7f9ae7"
            >
                        <button
                                    className="btn btn-default"
                                    data-test="copy-address"
                                    id="copy-to-clipboard"
                                    type="button"
                        >
                                    order.initial4
                        </button>
            </CopyToClipboard>
</div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="order.qr"
              className="qr"
              src="https://chart.googleapis.com/chart?chs=250x250&chld=L|2&cht=qr&chl=
			0x6ce146d0431079fe70dffee5fb15ba687d7f9ae7"
/>,
            "className": "qr-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "order.qr",
              "className": "qr",
              "src": "https://chart.googleapis.com/chart?chs=250x250&chld=L|2&cht=qr&chl=
			0x6ce146d0431079fe70dffee5fb15ba687d7f9ae7",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                order.initial1
                :
                 
                <span
                                className="time"
                >
                                <b />
                </span>
</h3>,
              <h4>
                order.initial2
                 
                <b>
                                0.10000000
                                 
                                ETH
                </b>
                 
                order.initial3
                <br />
                <b
                                className="address"
                                style={
                                                Object {
                                                                "wordWrap": "break-word",
                                                              }
                                }
                >
                                0x6ce146d0431079fe70dffee5fb15ba687d7f9ae7
                </b>
</h4>,
              <CopyToClipboard
                onCopy={[Function]}
                options={undefined}
                text="0x6ce146d0431079fe70dffee5fb15ba687d7f9ae7"
>
                <button
                                className="btn btn-default"
                                data-test="copy-address"
                                id="copy-to-clipboard"
                                type="button"
                >
                                order.initial4
                </button>
</CopyToClipboard>,
            ],
            "className": "details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "order.initial1",
                  ":",
                  " ",
                  <span
                    className="time"
>
                    <b />
</span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "order.initial1",
                ":",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <b />,
                    "className": "time",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "b",
                  },
                  "type": "span",
                },
              ],
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "order.initial2",
                  " ",
                  <b>
                    0.10000000
                     
                    ETH
</b>,
                  " ",
                  "order.initial3",
                  <br />,
                  <b
                    className="address"
                    style={
                                        Object {
                                                            "wordWrap": "break-word",
                                                          }
                    }
>
                    0x6ce146d0431079fe70dffee5fb15ba687d7f9ae7
</b>,
                ],
              },
              "ref": null,
              "rendered": Array [
                "order.initial2",
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "0.10000000",
                      " ",
                      "ETH",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "0.10000000",
                    " ",
                    "ETH",
                  ],
                  "type": "b",
                },
                " ",
                "order.initial3",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "br",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "0x6ce146d0431079fe70dffee5fb15ba687d7f9ae7",
                    "className": "address",
                    "style": Object {
                      "wordWrap": "break-word",
                    },
                  },
                  "ref": null,
                  "rendered": "0x6ce146d0431079fe70dffee5fb15ba687d7f9ae7",
                  "type": "b",
                },
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <button
                  className="btn btn-default"
                  data-test="copy-address"
                  id="copy-to-clipboard"
                  type="button"
>
                  order.initial4
</button>,
                "onCopy": [Function],
                "options": undefined,
                "text": "0x6ce146d0431079fe70dffee5fb15ba687d7f9ae7",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "order.initial4",
                  "className": "btn btn-default",
                  "data-test": "copy-address",
                  "id": "copy-to-clipboard",
                  "type": "button",
                },
                "ref": null,
                "rendered": "order.initial4",
                "type": "button",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
